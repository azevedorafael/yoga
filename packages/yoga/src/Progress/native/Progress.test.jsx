import React from 'react';
import { render } from '@testing-library/react-native';

import { ThemeProvider, Progress } from '../..';

describe('<Progress />', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <ThemeProvider>
        <Progress />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });
});
