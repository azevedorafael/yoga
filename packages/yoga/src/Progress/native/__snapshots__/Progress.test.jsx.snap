// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Progress /> should match snapshot 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    align="left"
    isNumber={false}
    style={
      Array [
        Object {
          "flexDirection": "column",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F5F5FA",
            "borderRadius": 8,
            "flexGrow": 1,
            "height": 8,
            "marginBottom": "auto",
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": "auto",
            "position": "relative",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#F46152",
              "borderRadius": 8,
              "height": "100%",
              "position": "absolute",
              "width": "NaN%",
            },
          ]
        }
        variant="primary"
        width={NaN}
      />
    </View>
  </View>
  <View
    align="left"
    isNumber={false}
    style={
      Array [
        Object {
          "flexDirection": "column",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F5F5FA",
            "borderRadius": 8,
            "flexGrow": 1,
            "height": 8,
            "marginBottom": "auto",
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": "auto",
            "position": "relative",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#F46152",
              "borderRadius": 8,
              "height": "100%",
              "position": "absolute",
              "width": "NaN%",
            },
          ]
        }
        variant="primary"
        width={NaN}
      />
    </View>
  </View>
  <View
    align="left"
    isNumber={false}
    style={
      Array [
        Object {
          "flexDirection": "column",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F5F5FA",
            "borderRadius": 8,
            "flexGrow": 1,
            "height": 8,
            "marginBottom": "auto",
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": "auto",
            "position": "relative",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#F46152",
              "borderRadius": 8,
              "height": "100%",
              "position": "absolute",
              "width": "60%",
            },
          ]
        }
        variant="primary"
        width={60}
      />
    </View>
  </View>
</View>
`;

exports[`<Progress /> should match snapshot with label number 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    align="left"
    isNumber={true}
    style={
      Array [
        Object {
          "flexDirection": "row-reverse",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F5F5FA",
            "borderRadius": 8,
            "flexGrow": 1,
            "height": 8,
            "marginBottom": "auto",
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": "auto",
            "position": "relative",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#F46152",
              "borderRadius": 8,
              "height": "100%",
              "position": "absolute",
              "width": "20%",
            },
          ]
        }
        variant="primary"
        width={20}
      />
    </View>
    <Text
      align="left"
      isNumber={true}
      style={
        Array [
          Object {
            "fontSize": 12,
            "marginRight": 8,
            "textAlign": "left",
            "width": 22,
          },
        ]
      }
    >
      20
    </Text>
  </View>
  <View
    align="right"
    isNumber={true}
    style={
      Array [
        Object {
          "flexDirection": "row",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F5F5FA",
            "borderRadius": 8,
            "flexGrow": 1,
            "height": 8,
            "marginBottom": "auto",
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": "auto",
            "position": "relative",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#F46152",
              "borderRadius": 8,
              "height": "100%",
              "position": "absolute",
              "width": "65%",
            },
          ]
        }
        variant="primary"
        width={65}
      />
    </View>
    <Text
      align="right"
      isNumber={true}
      style={
        Array [
          Object {
            "fontSize": 12,
            "marginLeft": 8,
            "textAlign": "right",
            "width": 22,
          },
        ]
      }
    >
      65
    </Text>
  </View>
</View>
`;

exports[`<Progress /> should match snapshot with label string 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    align="left"
    isNumber={false}
    style={
      Array [
        Object {
          "flexDirection": "column",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F5F5FA",
            "borderRadius": 8,
            "flexGrow": 1,
            "height": 8,
            "marginBottom": "auto",
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": "auto",
            "position": "relative",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#F46152",
              "borderRadius": 8,
              "height": "100%",
              "position": "absolute",
              "width": "NaN%",
            },
          ]
        }
        variant="primary"
        width={NaN}
      />
    </View>
    <Text
      align="left"
      isNumber={false}
      style={
        Array [
          Object {
            "fontSize": 12,
            "marginTop": 4,
            "textAlign": "left",
          },
        ]
      }
    >
      Some decription here
    </Text>
  </View>
  <View
    align="right"
    isNumber={false}
    style={
      Array [
        Object {
          "flexDirection": "column",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F5F5FA",
            "borderRadius": 8,
            "flexGrow": 1,
            "height": 8,
            "marginBottom": "auto",
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": "auto",
            "position": "relative",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#F46152",
              "borderRadius": 8,
              "height": "100%",
              "position": "absolute",
              "width": "65%",
            },
          ]
        }
        variant="primary"
        width={65}
      />
    </View>
    <Text
      align="right"
      isNumber={false}
      style={
        Array [
          Object {
            "fontSize": 12,
            "marginTop": 4,
            "textAlign": "right",
          },
        ]
      }
    >
      Some decription here
    </Text>
  </View>
</View>
`;

exports[`<Progress /> should match snapshot with secondary variant prop 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    align="left"
    isNumber={true}
    style={
      Array [
        Object {
          "flexDirection": "row-reverse",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F5F5FA",
            "borderRadius": 8,
            "flexGrow": 1,
            "height": 8,
            "marginBottom": "auto",
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": "auto",
            "position": "relative",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#652CDB",
              "borderRadius": 8,
              "height": "100%",
              "position": "absolute",
              "width": "20%",
            },
          ]
        }
        variant="secondary"
        width={20}
      />
    </View>
    <Text
      align="left"
      isNumber={true}
      style={
        Array [
          Object {
            "fontSize": 12,
            "marginRight": 8,
            "textAlign": "left",
            "width": 22,
          },
        ]
      }
    >
      20
    </Text>
  </View>
</View>
`;

exports[`<Progress /> should match snapshot with tertiary variant prop 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    align="left"
    isNumber={true}
    style={
      Array [
        Object {
          "flexDirection": "row-reverse",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F5F5FA",
            "borderRadius": 8,
            "flexGrow": 1,
            "height": 8,
            "marginBottom": "auto",
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": "auto",
            "position": "relative",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#2B9486",
              "borderRadius": 8,
              "height": "100%",
              "position": "absolute",
              "width": "20%",
            },
          ]
        }
        variant="tertiary"
        width={20}
      />
    </View>
    <Text
      align="left"
      isNumber={true}
      style={
        Array [
          Object {
            "fontSize": 12,
            "marginRight": 8,
            "textAlign": "left",
            "width": 22,
          },
        ]
      }
    >
      20
    </Text>
  </View>
</View>
`;
